cli/src/main/scala/com/advancedtelematic/tuf/cli/repo/TufRepo.scala

```diff
@@ -295,8 +296,15 @@ abstract class TufRepo[S <: TufServerClient](val repoPath: Path)(implicit ec: Ex
     signatures
       .map { case (key, sig) => (keyStorage.readPublicKey(key), sig) }
       .sequence
-      .map { sigs => // TODO: RSASSA_PSS_SHA256 hard coded here. This is not good enough.
-        sigs.transform { case (key, sig) => ClientSignature(key.id, RSASSA_PSS_SHA256, sig) }
+      .map { sigs =>
+        sigs.transform { case (key, sig) => 
+          // Choose signature method based on key type
+          val method = key.keytype match {
+            case Ed25519KeyType => ED25519 
+            case _ => RSASSA_PSS_SHA256
+          }
+          ClientSignature(key.id, method, sig)
```

Comment:
You don't need to match, this is actually available using the keytype:

```
        sigs.transform { case (key, sig) => ClientSignature(key.id, key.keytype.crypto.signatureMethod, sig) }
```

Reply:
Oh, nice. Fixed, removed the now-unused imports.